### PR TITLE
DYN-3241: Deploy md2html.exe as a single.exe

### PR DESCRIPTION
**Pull request by sm6srw**
### Purpose

Deploy md2html.exe as a single.exe

This Pull Request does:
* deploy md2html.exe as a single exe in windows
* keep the old behavior for mono.

Note: I did not find a good way to use the same project file for both windows and mono. Let me know if you know about a way to detect mono inside a csproj file.

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

@mjkkirschner

### CODE REVIEW

File: src/Tools/Md2Html/Md2Html.csproj [hunk 1]
```diff
@@ -69,4 +74,21 @@
     </PackageReference>
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+  <Target Name="AfterBuild">
+    <!-- the ILMergePath property points to the location of ILMerge.exe console application -->
+    <Exec Command="$(ILMergeConsolePath) $(OutputPath)Md2Html.exe ^
```
Comment (mjkkirschner) [inline src/Tools/Md2Html/Md2Html.csproj]: where does this live though? In the packages folder?

File: src/Tools/Md2Html/Md2Html.csproj [hunk 2]
```diff
@@ -30,7 +31,8 @@
     <PlatformTarget>AnyCPU</PlatformTarget>
     <DebugType>pdbonly</DebugType>
     <Optimize>true</Optimize>
-    <OutputPath>$(OutputPath)\Md2Html\</OutputPath>
+    <SharedOutputPath>$(OutputPath)\Md2Html\</SharedOutputPath>
+    <OutputPath>bin\AnyCPU\Release\</OutputPath>
```
Comment (mjkkirschner) [inline src/Tools/Md2Html/Md2Html.csproj:35]: can you explain this and why it needs two output paths and hardcoded path?

**Comment by mjkkirschner** [comment]
@sm6srw did you try `'$(OS)` ? that should let you detect the OS in msbuild and decide to invoke the script or not?

File: src/Tools/Md2Html/Md2Html.csproj (hunk 1 above)
Comment (sm6srw) [reply src/Tools/Md2Html/Md2Html.csproj]: Yes, that just works.

File: src/Tools/Md2Html/Md2Html.csproj (hunk 2 above)
Comment (sm6srw) [reply src/Tools/Md2Html/Md2Html.csproj:35]: `SharedOutputPath` points to a `md2Html` directory in our shared bin folder.
`OutputPath` points to a local output folder relative to the project file.

`ILMerge` is then used for building a single .exe that is placed in the shared folder from the files in the local bin folder.

**Comment by sm6srw** [comment]
Yes, that might work, assuming nobody uses mono on windows? I will take a look.

**Comment by mjkkirschner** [comment]
@sm6srw it looks good either way - but maybe one improvement would be to add a local readme file to this folder describing the build steps and super briefly what the tool does?

**Comment by sm6srw** [comment]
@mjkkirschner Reusing the same csproj file now. I will add a README for the tool as a separate task.

**Comment by sm6srw** [comment]
Added Readme. I will merge this PR now.

**Pull request merged**